lib/node_modules/@stdlib/assert/is-domain-name/lib/main.js

```diff
@@ -0,0 +1,110 @@
+/*
+* @license Apache-2.0
+*
+* Copyright (c) 2022 The Stdlib Authors.
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*    http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+'use strict';
+
+// MODULES //
+
+var isString = require( '@stdlib/assert/is-string' ).isPrimitive;
+
+
+// VARIABLES //
+
+/**
+* Regular expression: `/^(?:[a-z0-9](?:[a-z0-9-]{0,61}[a-z0-9])?\.)+[a-z0-9][a-z0-9-]{0,61}[a-z0-9]$/`
+*
+* -   `\^`
+*     -   match any string which begins with
+*
+* -   `(?:)`
+*     -   capture but do not remember
+*
+* -   `[a-z0-9]`
+*     -   match a single digit or letter
+*
+* -   `(?:)`
+*     -   capture but do not remember
+*
+* -   `[a-z0-9-]{0,61}`
+*     -   match between 0 and 61 digits, letters, or hyphens
+*
+* -   `[a-z0-9]`
+*     -   match a single digit or letter
+*
+* -   `?`
+*     -   match the previous token between zero and one times
+*
+* -   `\.`
+*     -   match a literal `.`
+*
+* -   `+`
+*     -   match the previous group one or more times
+*
+* -   `[a-z0-9]`
+*     -   match a single digit or letter
+*
+* -   `[a-z0-9-]{0,61}`
+*     -   match between 0 and 61 digits, letters, or hyphens
+*
+* -   `[a-z0-9]`
+*     -   match a single digit or letter
+*
+* -   `$`
+*     -   match end of string
+*/
```

Comment:
```suggestion
*
* @private
* @constant
* @type {RegExp}
*/
```

As this is a JSDoc comment, we need to add the `private` tag.